src/BlobAdvanced.java

```diff
@@ -88,273 +95,258 @@ void runSamples() throws InvalidKeyException, URISyntaxException, IOException {
-            if (container.deleteIfExists()) {
-                System.out.println(String.format("Successfully deleted the container: %s", container.getName()));
+        } finally {
+            if (containerClient.exists()) {
+                containerClient.delete();
+                System.out.println(String.format("Successfully deleted the container: %s", containerClient.getBlobContainerName()));
             }
         }
     }
 
     /**
      * Manage container metadata
-     * @param blobClient Azure Storage Blob Service
+     *
+     * @param blobServiceClient Azure Storage Blob Service
      */
-    private void containerMetadata(CloudBlobClient blobClient) throws URISyntaxException, StorageException {
+    private void containerMetadata(BlobServiceClient blobServiceClient) throws URISyntaxException, BlobStorageException {
         // Get a reference to a container
         // The container name must be lower case
-        CloudBlobContainer container = blobClient.getContainerReference("blobadvancedcontainer"
-                + UUID.randomUUID().toString().replace("-", ""));
+        String containerName = "blobadvancedcontainer" + UUID.randomUUID().toString().replace("-", "");
+        BlobContainerClient containerClient = blobServiceClient.getBlobContainerClient(containerName);
         try {
             System.out.println("Create container");
             // Create the container if it does not exist
-            container.createIfNotExists();
-
+            if (!containerClient.exists()) {
+                containerClient.create();
+            }
             System.out.println("Set container metadata");
-            container.getMetadata().put("key1", "value1");
-            container.getMetadata().put("foo", "bar");
-            container.uploadMetadata();
-
+            Map<String, String> metadataMap = containerClient.getProperties().getMetadata();
+            metadataMap = metadataMap != null ? metadataMap : new HashMap();
+            metadataMap.put("key1", "value1");
+            metadataMap.put("foo", "bar");
+            containerClient.setMetadata(metadataMap);
             System.out.println("Get container metadata:");
-            HashMap<String, String> metadata = container.getMetadata();
-            Iterator it = metadata.entrySet().iterator();
-            while (it.hasNext()) {
-                Map.Entry pair = (Map.Entry) it.next();
+            Map<String, String> metadata = containerClient.getProperties().getMetadata();
+            metadata.entrySet().forEach(pair -> {
                 System.out.printf(" %s = %s%n", pair.getKey(), pair.getValue());
-                it.remove();
-            }
-        }
-        finally {
-            if (container.deleteIfExists()) {
-                System.out.println(String.format("Successfully deleted the container: %s", container.getName()));
+            });
+            metadata.clear();
+        } finally {
+            if (containerClient.exists()) {
```

Comment:
As mentioned above, the preferred option is trying to delete first and then catching and handling a potential 404